public/app/percona/entitlements/EntitlementsPage.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import PageWrapper from '../shared/components/PageWrapper/PageWrapper';
+import { useStyles2, Spinner } from '@grafana/ui';
+import { LIST_ENTITLEMENTS_CANCEL_TOKEN, PAGE_MODEL } from './Entitlements.contants';
+import { logger } from '@percona/platform-core';
+import { useSelector } from 'react-redux';
+import { CollapsableSection } from '@grafana/ui/src/components';
+import { useCancelToken } from '../shared/components/hooks/cancelToken.hook';
+import { isApiCancelError } from '../shared/helpers/api';
+import { getStyles } from './Entitlements.styles';
+import EntitlementsService from './Entitlements.service';
+import { Entitlement } from './Entitlements.types';
+import { SectionContent } from './components/SectionContent/SectionContent';
+import { Label } from './components/SectionLabel/SectionLabel';
+import { PlatformConnectedLoader } from '../shared/components/Elements/PlatformConnectedLoader';
+import { StoreState } from 'app/types';
+
+const EntitlementsPage: FC = () => {
+  const [pending, setPending] = useState(true);
+  const [data, setData] = useState<Entitlement[]>([]);
+  const isConnectedToPortal = useSelector((state: StoreState) => !!state.perconaUser.isConnectedToPortal);
+  const [generateToken] = useCancelToken();
+  const styles = useStyles2(getStyles);
+
+  const getData = useCallback(async (showLoading = false) => {
+    showLoading && setPending(true);
+
+    try {
+      const entitlements = await EntitlementsService.list(generateToken(LIST_ENTITLEMENTS_CANCEL_TOKEN));
```

Comment:
Shouldn't this be done with an action in the store?

Reply:
What is reason for putting this to the store when we are using it just here and inside nested components related to this?

Reply:
Sorry this was my confusion. (though this was portal project for a bit 😅)